### DISCUSSION TITLE
Can Web run without the Babel plugin?

### DISCUSSION THREAD

**Discussion by nandorojo** [software-mansion/react-native-reanimated#2620] [Q&A]
Since Web is JS-only, I think it would be best to find a way to get it to run without the Babel plugin. The plugin adds a lot of build time, and makes it hard to use with other build tools like Next.js.

**Comment by louisgv** [comment]
Another way of approaching this imo, is maybe to have the abstract layer of reanimated (in this context, I thought of @nandorojo 's moti library) to bundle the js portion of reanimated when serving for web and lock the version (the same way expo lock the version of their modules) 

Or have another abstraction library in between RA2 and moti that has proper standalone web bundling.

**Reply by nandorojo** [reply]
I would be open to this too

**Reply by wibb36** [reply]
if using Expo? @expo/next-adapter is another one that rely on Babel. How do you go about that?

**Reply by nandorojo** [reply]
If you aren’t using Reanimates or Moti, then you can actually use Next.js + RN / Expo without Babel. Kind of crazy - it just works.

**Reply by louisgv** [reply]
@nandorojo would be down to investigate some approach to this. It would be very helpful if we can use moti without having to do the "dangerously bundle" webpack config stuff, but rather just have it all bundled and good to go. From what I've seen so far, swcpack seems interesting but still unstable, parcel is using swc underneath and it's pretty cool (I used it on a playground project before)

One idea I have right now is to simply do an experimentation repo with parcel to bundle reanimated for web just to have a POC build path. Then maybe take that POC to moti or just do quick integration. I don't mind abstraction, but maintenance might be a drag - tho if it's just a bundling process in between, it might not be that bad? 

The other concern is whether this bundled module would play well with Expo - Moti - RA2 on the native side. I think if Moti just do platform specific codepath to pick the bundled RA2, then we likely not have to worry about it at all.

**Reply by nandorojo** [reply]
After more investigation, it seems like the babel plugin is needed even on web. It appends a `_closure` to every worklet, telling it each variable to track, which it then uses in the worklet's dependencies. There *may* be a way around this, if all of your worklets/hooks have a dependency array declared, but I haven't tested it.

**Reply by louisgv** [reply]
Would it be possible for Moti to abstract away the worklet dependencies on web? I wouldn't mind the potential constraint (i.e, Moti defines a specific interface for animating, which are always passed down to worklet) if it can provide frictionless animation cross-platform.

**Reply by nandorojo** [reply]
This is currently the case on canary: https://github.com/nandorojo/moti/pull/121

Still investigating if this is sufficient
